packages/react/src/transactions/adapters/responders/CreateMirrorResponder.ts

```diff
@@ -1,25 +1,47 @@
 import {
+  getAllPendingTransactions,
   GetPublicationData,
   GetPublicationDocument,
   GetPublicationVariables,
   getSession,
+  isMirrorTransactionFor,
+  PublicationStats,
   SafeApolloClient,
   SessionType,
-  Sources,
+  Sources
 } from '@lens-protocol/api-bindings';
 import { CreateMirrorRequest } from '@lens-protocol/domain/use-cases/publications';
 import {
   ITransactionResponder,
   TransactionData,
 } from '@lens-protocol/domain/use-cases/transactions';
+import { invariant } from '@lens-protocol/shared-kernel';
 
 export class CreateMirrorResponder implements ITransactionResponder<CreateMirrorRequest> {
   constructor(private readonly client: SafeApolloClient, private readonly sources: Sources) {}
 
+  async prepare({ request }: TransactionData<CreateMirrorRequest>) {
+    const identifier = this.client.cache.identify({
+      __typename: 'Post',
+      id: request.publicationId,
+    });
+    this.client.cache.modify({
+      id: identifier,
+      fields: {
+        stats(existing: PublicationStats) {
+          return {
+            ...existing,
+            totalAmountOfMirrors: existing.totalAmountOfMirrors + 1,
+          };
+        },
+      },
+    });
```

Comment:
The challenge with these programmatic cache updates and the reason shy we started moving away from them and using more field policies is that they rely on the cache to be primed with the node we are aiming to update (a Publication in this case). This works in the immediate of a UI but does not work well in another thread tha tmight not have this exact item in cache (e.g. another tab) or during hard refreshes where the SDK bootstrap by being pure JS logic tries to update cache before any active query is able to fetch the relevant item.